safe-hts-precompile/SafeHTS.sol

```diff
@@ -0,0 +1,458 @@
+// SPDX-License-Identifier: Apache-2.0
+pragma solidity >=0.5.0 <0.9.0;
+pragma experimental ABIEncoderV2;
+
+import "../hts-precompile/IHederaTokenService.sol";
+import "./HederaResponseCodes.sol";
+
+library SafeHTS {
+
+    address constant precompileAddress = address(0x167);
+    // 90 days in seconds
+    uint32 constant defaultAutoRenewPeriod = 7776000;
+
+    function safeCryptoTransfer(IHederaTokenService.TokenTransferList[] memory tokenTransfers) internal {
+        int32 responseCode;
+        (bool success, bytes memory result) = precompileAddress.call(
+            abi.encodeWithSelector(IHederaTokenService.cryptoTransfer.selector, tokenTransfers));
+        responseCode = success ? abi.decode(result, (int32)) : HederaResponseCodes.UNKNOWN;
+        require(responseCode == HederaResponseCodes.SUCCESS, "Safe crypto transfer failed!");
+    }
+
+    function safeMintToken(address token, uint64 amount, bytes[] memory metadata) internal
+    returns (uint64 newTotalSupply, int64[] memory serialNumbers) {
+        int32 responseCode;
+        (bool success, bytes memory result) = precompileAddress.delegatecall(
```

Comment:
Do we deliberately use `delegatetcall` here?

Reply:
It was not added in the commits made by me, but looking at the [hedera-services PR](https://github.com/hashgraph/hedera-services/pull/2866) that precedes this PR it seems to be fixed there.

I don't see an obvious reason to use `delegatecall` here so I will change it to `call`.